File: src/values/i8.rs
pub struct ValueI8 {
    value: Option<i8>,
}

impl ValueI8 {
    pub fn init_empty() -> ValueI8 {
        return ValueI8 { value: None };
    }

    pub fn init_with(value: i8) -> ValueI8 {
        return ValueI8 { value: Some(value) };
    }
    
    pub fn show(&self, end: char) {
        if let Some(val) = self.value {
            print!("{val}{end}")
        } else {
            print!("null{end}");
        }
    }
    
    pub fn is_null(&self) -> bool {
        return self.value.is_none();
    }
    
    pub fn get_type_representation(&self) -> &str {
        return "i8";
    }
    
    pub fn get_value(&self) -> Option<i8> {
        return self.value;
    }
}

File: src/values/i8.cpp
#include "values/i8.hpp"
using namespace common::values;

ValueI8::ValueI8()
    : value(std::nullopt) {
    
}

ValueI8::ValueI8(int8_t value)
    : value(value) {
    
}

void ValueI8::show(char end) const {
    if (this->value)
        std::cout << "u8<" << (int16_t)this->value.value() << '>' << end;
    else
        std::cout << "u8<null>" << end;
}

bool ValueI8::isNull() const {
    return !this->value.has_value();
}

std::string ValueI8::getTypeRepr() const {
    return "i8";
}

ValueKind ValueI8::getKind() const {
    return ValueKind::I8;
}

std::optional<int8_t> ValueI8::getValue() const {
    return this->value;
}